### ISSUE TITLE
If you click on a country(second time) and the date indicator is on, corona closes

### ISSUE THREAD

**Issue by gcarreno** [wp-xyz/corona#42]
Hey Werner(@wp-xyz),

When I was facepalming while trying to see what you mentioned on #41, I bumped into this sequence:
1. Open corona
2. Turn Date Indicator ON
3. Click on any country
4. Corona exits

Then I tried some other stuff and also got abrupt close on this sequence:
1. Open corona
2. Select country from tree
3. Turn Date Indicator ON
4. Move map slider somewhere that is not max date
5. Select another country on the tree
6. Corona exits

Another way of reproducing it:
1. Open corona
2. Click on country
3. Turn Date Indicator ON
4. Click on another country
5. Corona exits

So, in conclusion, when you turn Date Indicator ON and you select another country, :bomb:  BOOM :bomb: 

Here's the heaptrace:
```
WARNING: TChartListbox.Destroy with LCLRefCount>0. Hint: Maybe the component is processing an event?
WARNING: TTreeView.Destroy with LCLRefCount>0. Hint: Maybe the component is processing an event?
[TGtk2WidgetSet.Destroy] WARNING: There are 1 unreleased DCs, a detailed dump follows:
[TGtk2WidgetSet.Destroy]  DCs:   00007F2AA694FC40
[TGtk2WidgetSet.Destroy] WARNING: There are 1 unreleased GDIObjects, a detailed dump follows:
[TGtk2WidgetSet.Destroy]   GDIOs: 00007F2AA67AE6C0
[TGtk2WidgetSet.Destroy]   gdiBrush: 1
```
Here's my `corona.cfg`:
[corona.cfg.zip](https://github.com/wp-xyz/corona/files/6068746/corona.cfg.zip)

Hope it helps!

Cheers,
Gus

**Comment by wp-xyz** [comment]
I cannot reproduce these issues (which I have seen in a different context, though). The problem is that the program stores the state of the last session. 

For this issue, let's start in "virgin" state. In menu "configuration", uncheck "Automatic saving/loading"; delete the cfg file and the data files. After the second start (when cfg has been recreated) the program should behave reproducibly. Play with it, but without data. Hopefully there should be no crashes any more (I fixed a couple of them a minute now).

Then download the data. Play again. When a crash happens, restart the program and repeat the steps you took. If you can reproduce the issue, report it precisely with the steps taken.

Later we can continue with active cfg file. In this case, you should probably post your cfg file; hopefully my app will behave like yours when I copy your cfg over mine.

**Comment by gcarreno** [comment]
Hi Werner(@wp-xyz),

> I cannot reproduce these issues (which I have seen in a different context, though). The problem is that the program stores the state of the last session.

Even with the `*.cfg` file that I provided?
That is quite strange, and I'll tell you why: I had to `rm -rf` my corona cloned repo because `source/cabout.pas` had some hickup and was stuck on either `CRLF` or `LF`, so I cloned the repo fresh and compiled it from 3 or 4 commits ago.

It should be as pristine/virgin as you asked. But none the less, let me pull the latest and _re-virginate_ all the things and report back.

Cheers,
Gus

**Comment by gcarreno** [comment]
Hey Werner(@wp-xyz),

How much are you willing to bet that if you close #43 this one goes away?

Cheers,
Gus

**Comment by gcarreno** [comment]
Hey Werner(@wp-xyz),

Nevermind, I solved the maps not being compiled but this issue continues.

Cheers,
Gus

**Comment by wp-xyz** [comment]
Did not notice your cfg file. But even when I use it I cannot reproduce the issue. (BTW, I now split the "Automatic load/save" menu item into separate "Automatic load" and "Automatic save" because when the action was checked previously to read the cfg file at program start it would automatically write it at program start and thus overwrite the cfg. Now I can handle your case with enabled auto-loading and disabled auto-saving).

Since you mentioned the map resources: Resource files always have been kind of tricky, and I cannot guarantee that the new map resources are automatically included. Please try a "Run" > "Build" from Lazarus which I know to rewrite the res file.

**Comment by gcarreno** [comment]
Hey Werner(@wp-xyz),

Something strange is happening on my side for sure.
I've now compiled from the command line, under the source folder, these ways:
1. `lazbuild -B corona.lpi`; This uses the Default build mode still :bomb: 
2. `lazbuild -B --bm=Debug corona.lpi`; still :bomb: but with loads of more info dump
3. `lazbuild -B --bm=Release corona.lpi`; still :bomb: 

And from the IDE:
1. `Shift+F9`(Build) and `Ctrl+Shift+F9`(Run no Debug); does not :bomb: but the Date Indicator is turned off every time I select another country.

This is some voodoo magic here!! The bug only shows when compiled with lazbuild?!?! This is really freaky!!

Cheers,
Gus

**Comment by gcarreno** [comment]
Hi Werner(@wp-xyz),

**EXPLETIVE**!! **EXPLETIVE**!! **EXPLETIVE**!! **EXPLETIVE**!! **EXPLETIVE**!! **EXPLETIVE**!!

I'm SOOO sorry for wasting your time!!!

Just found out that `lazbuild` trunk and 2.0.6 have no issue, but `lazbuild` 2.0.[12,10] are the ones that :bomb:s
I can reproduce on both inside the IDE.
So it's definitely a problem with FPC 3.2.0!! Progress.... I hope!!

The problem is on line 827 of cMain.pas:
```pascal
  Chart.ClearSeries;
```

So whatever you're doing by adding the purple line to the chart, PFC 3.2.0 does not like when you tell Chart to clear the series.

I'm feeling really tired, since I've been up for far too long. I'll debug this tomorrow, or my type of tomorrow: after I've slept enough.

Cheers,
Gus

**Comment by wp-xyz** [comment]
Fixed. A clear indication of poor design of the TChartListBox component (done by myself...): Chart.ClearSeries destroys the series one by one, and each time TChartListbox is notified to rebuild itself. During this process the listbox checks whether the series DateIndicatorLine is nil -- and of course, DateIndicatorLine is soon destroyed but it is not reset to nil...

**Comment by gcarreno** [comment]
Hi Werner(@wp-xyz),

I'm glad that you found the issue!! And we are human after all, so we are prone to error :smile: 

What really baffles me is the fact that FPC 3.0.4 and FPC 3.3.1 just gloss over that issue, hein?!

Cheers,
Gus

**Comment by gcarreno** [comment]
Closed by 624e4c8
